pkg/cloudinit/cloudinit.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2022 SUSE.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudinit
```

Comment:
We should consider moving this package (and others) to `internal` as they are really only for use within this provider and not externally.

It also has added benefit that we can then run apidiff.

Reply:
Done!